prisma/schema.prisma

```diff
@@ -1947,6 +1998,163 @@ model HomeBlock {
   clones    HomeBlock[]   @relation("Clones")
 }
 
+enum Currency {
+  USD
+  BUZZ
+}
+
+enum BountyType {
+  ModelCreation
+  LoraCreation
+  EmbedCreation
+  DataSetCreation
+  DataSetCaption
+  ImageCreation
+  VideoCreation
+  Other
+}
+
+enum BountyMode {
+  Individual
+  Split
+}
+
+enum BountyEntryMode {
+  Open
+  BenefactorsOnly
+}
+
+model Bounty {
+  id                      Int             @id @default(autoincrement())
+  userId                  Int?
+  user                    User?           @relation(fields: [userId], references: [id], onDelete: SetNull)
+  name                    String // Locked after created
+  description             String
+  startsAt                DateTime
+  expiresAt               DateTime // Locked after created
+  createdAt               DateTime        @default(now())
+  updatedAt               DateTime        @updatedAt
+  details                 Json?
+  mode                    BountyMode      @default(Individual)
+  entryMode               BountyEntryMode @default(Open)
+  type                    BountyType
+  minBenefactorUnitAmount Int
+  maxBenefactorUnitAmount Int? // Default to initial benefactor's entry
+  entryLimit              Int             @default(1)
+  nsfw                    Boolean         @default(false)
+  poi                     Boolean         @default(false)
+  complete                Boolean         @default(false)
+
+  tags        TagsOnBounty[]
+  entries     BountyEntry[]
+  benefactors BountyBenefactor[]
+  engagements BountyEngagement[]
+  thread      Thread?
+  metrics     BountyMetric[]
+  rank        BountyRank?
+  stats       BountyStat?
+  reports     BountyReport[]
+}
+
+model BountyEntry {
+  id        Int      @id @default(autoincrement())
+  userId    Int?
+  user      User?    @relation(fields: [userId], references: [id], onDelete: SetNull)
+  bountyId  Int
```

Comment:
We would for sure want to put an index on bountyId at least so that we can efficiently lookup entries for bounties.